pytorch_to_returnn/converter/converter.py

```diff
@@ -168,10 +168,10 @@ def _make_tf_feed_dict(self, input: Data, out_entry: Optional[TensorEntry] = Non
         visited_entries.add(entry_)
         if not all(call_.module.module.is_deterministic for call_ in entry_.output_from_calls):
           non_deterministic_tensor_entries.append(entry_)
+          continue
         for call_ in entry_.output_from_calls:
-          for prev_entry_ in [pe for pe in call_.inputs_args if isinstance(pe, TensorEntry)]:
-            if prev_entry_ not in visited_entries:
-              entries_to_visit.append(prev_entry_)
+          for prev_entry_ in [pe for pe in call_.inputs_tensor_deps if pe not in visited_entries]:
```

Comment:
As said, I would not create such a temp list here. You can simply move any extra checks into the loop.